reps/2024-06-13-export-api.md

```diff
@@ -0,0 +1,153 @@
+## Summary
+### General Motivation
+
+In the current design of Ray, the way to export various states in the Ray cluster are inconsistent.
+For example, the state of the Actor is broadcast through GCS pubsub, and to obtain the state change of the node,
+it is necessary to query rpc service (NodeInfoGcsService). For the jobs submitted through job submission,
+there is no way to expose the state. The high-level libraries on top of Ray also don't have a unified export way.
+E.g. RayServe and RayData collect the states through their own StateActor and report to the Dashboard respectively.
+
+It is very difficult to obtain these ray basic states outside the Ray cluster. This issue is reflected in two aspects:
+1. Scale of data exceeds current dashboard API limits.
+2. After the Ray cluster terminates, all status data is lost.
+If a unified export API can be defined,
+we can achieve the observable ability independent of the Ray cluster. The most typical scenario is to build the Ray history server.
```

Comment:
I would re-organize this to something like 
```suggestion
There are several major improvements requested by users about the Ray Dashboard
1. Currently, all state data is lost when the Ray cluster terminates and users are looking for persistence of this data.
2. The Ray dashboard has scalability limits on the amount of data stored and returned. Advanced users would benefit from greater scalability.
3. There is no unified and accurate way to get state events pushed on state change across various resource types. The State and dashboard APIs support a pull model, where it is possible to miss some events. There is also not a consistent way across resource types to get this information directly from components that generate it.

A unified export API could serve as a base to address these issues by allowing state data to be exported and processed separately. This would ensure that observability features remain functional independent of the Ray cluster.
```

Reply:
> to obtain the state change of the node, it is necessary to query rpc service (NodeInfoGcsService)

Just curious, do you or other users already query these GCS services directly to get data?